### PR TITLE
Navegg UserId: add new userid submodule

### PR DESCRIPTION
**Pull request by hugopenha-navegg**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by hugopenha-navegg** [comment]
changes to add an navegg userid

**Comment by jeanstemp** [comment]
Docs PR: https://github.com/prebid/prebid.github.io/pull/3080

**Comment by hugopenha-navegg** [comment]
The documentation is ok and I believe the code is too. Can it be revised again?

**Comment by jlukas79** [comment]
Missing unit tests

**Comment by hugopenha-navegg** [comment]
Added unit test

File: modules/.submodules.json [hunk 1]
```diff
@@ -28,9 +28,9 @@
     "uid2IdSystem",
     "admixerIdSystem",
     "dmdIdSystem",
-    "akamaiDAPId",
```
Comment (ncolletti) [inline modules/.submodules.json:31]: this should not be removed

File: modules/.submodules.json [hunk 2]
```diff
@@ -44,7 +44,6 @@
     "haloRtdProvider",
     "iasRtdProvider",
     "jwplayerRtdProvider",
-    "medianetRtdProvider",
```
Comment (ncolletti) [inline modules/.submodules.json:47]: this should also not be removed

File: modules/naveggId.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ */
+import * as utils from '../src/utils.js'
+import { submodule } from '../src/hook.js';
+import { getStorageManager } from '../src/storageManager.js';
+
+const MODULE_NAME = 'naveggId';
+export const storage = getStorageManager();
+
+function readnaveggIdFromLocalStorage() {
+  return storage.localStorageIsEnabled ? storage.getDataFromLocalStorage('nvggid') : null;
```
Comment (ncolletti) [inline modules/naveggId.js:15]: storage.localStorageIsEnabled can likely be run once in your getId function and removed from each of these helper functions.

File: modules/naveggId.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+function readnaveggIDFromCookie() {
+  return storage.cookiesAreEnabled ? storage.getCookie('nvggid') : null;
+}
+
+function readoldnaveggIDFromCookie() {
+  return storage.cookiesAreEnabled ? storage.getCookie('nid') : null;
```
Comment (ncolletti) [inline modules/naveggId.js:23]: would be easier to read if you stored const variables and referenced them here: const OLD_NAVEGG_ID = 'nid' and const NAVEGG_ID = 'nvggid'

File: modules/naveggId.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    * @param {SubmoduleConfig} config
+    * @return {{id: string | undefined } | undefined}
+    */
+  getId() {
+    const naveggIdString = readnaveggIdFromLocalStorage() || readnaveggIDFromCookie() || readoldnaveggIDFromCookie() || readnvgIDFromCookie() || readnavIDFromCookie() || readnvgnavFromLocalStorage();
+
+    if (typeof naveggIdString == 'string' && naveggIdString) {
+      try {
+        return { id: naveggIdString };
+        //        return { id: naveggIdString ? { naveggId: naveggIdString } : undefined }
```
Comment (ncolletti) [inline modules/naveggId.js:75]: this can be removed

**Comment by ncolletti** [comment]
- Please add Unit Tests. 
- Do you plan on adding a GVLID for GDPR compatibility? It is recommended to include if you plan on supporting EU. 
- Your file should be renamed from `naveggId` to `naveggIdSystem`
- It looks like your id module is not saving an Id to the browser at any point, to confirm is this intended and you are syncing outside of Prebid or are you intending to add this functionality?
- You need to add an entry in userId.md and userId_example.html

**Comment by ncolletti** [comment]
Would be good to include a unit test for your getId function. Also missing your entry in userid_example.html

**Comment by hugopenha-navegg** [comment]
> Would be good to include a unit test for your getId function. Also missing your entry in userid_example.html



> * Please add Unit Tests.
> * Do you plan on adding a GVLID for GDPR compatibility? It is recommended to include if you plan on supporting EU.
> * Your file should be renamed from `naveggId` to `naveggIdSystem`
> * It looks like your id module is not saving an Id to the browser at any point, to confirm is this intended and you are syncing outside of Prebid or are you intending to add this functionality?
> * You need to add an entry in userId.md and userId_example.html

Do you plan on adding a GVLID for GDPR compatibility? It is recommended to include if you plan on supporting EU.

Resp.: We plan to include, but as of now we are not operating in EU, only in South America.

It looks like your id module is not saving an Id to the browser at any point, to confirm is this intended and you are syncing outside of Prebid or are you intending to add this functionality?

TCV: This is intentional, as our ID is already saved. We are getting the ID and transiting it over prebid process. So in a way we are syncing it ourside of Prebid.

**Comment by hugopenha-navegg** [comment]
> Would be good to include a unit test for your getId function. Also missing your entry in userid_example.html

A test was performed to check if the naveggId exists or not, which test is needed to be ok?

**Comment by ncolletti** [comment]
Include coverage for your getId function, see quantcast and mwopenlink spec files for examples.

**Comment by hugopenha-navegg** [comment]
The last build failed. It seems to be in the BrowserStack with IE 11. Would you know how to tell me why it failed?

**Review by ncolletti — APPROVED** [review]
LGTM

**Comment by ncolletti** [comment]
> The last build failed. It seems to be in the BrowserStack with IE 11. Would you know how to tell me why it failed?

Seems like a flaky IE test that is affecting other PRs as well, unrelated to your code.

**Comment by hugopenha-navegg** [comment]
> > The last build failed. It seems to be in the BrowserStack with IE 11. Would you know how to tell me why it failed?
> 
> Seems like a flaky IE test that is affecting other PRs as well, unrelated to your code.

Okay, so the error related to IE is not an impediment for the pull request to be approved, right?

**Comment by bretg** [comment]
We need another reviewer. @jlukas79 has agreed to take a look

File: modules/userId/eids.md [hunk 6]
```diff
@@ -182,13 +190,6 @@ userIdAsEids = [
             atype: 3
         }]
     },
```
Comment (jlukas79) [inline modules/userId/eids.md:192]: why is the akamai block being removed, akamaiDAPId is still valid module

**Review by jlukas79 — CHANGES_REQUESTED** [review]
Looks good after the akamai comment is resolved

**Comment by hugopenha-navegg** [comment]
> Looks good after the akamai comment is resolved

Done

File: modules/userId/eids.md [hunk 7]
```diff
@@ -182,13 +190,6 @@ userIdAsEids = [
...
-    {
-        source: 'akamai.com',
```
Comment (jlukas79) [inline modules/userId/eids.md:186]: This akamai block should not be removed from eids.md. All other files are fine

Comment (hugopenha-navegg) [reply modules/userId/eids.md:186]: I believe it was a mistake, I added again.

**Review by jlukas79 — APPROVED** [review]

**Comment by hugopenha-navegg** [comment]
Do I need to do anything else? Could you tell me when it will be sent to master?

**Pull request merged**